Dockerfile-linux

```diff
@@ -0,0 +1,53 @@
+FROM ubuntu:xenial
+MAINTAINER Pranjal <pranjal@brave.com>
+
+RUN \
+  apt-get update && \
+  apt-get install -y build-essential curl autotools-dev automake gnupg2 && \
+  curl -fsSL "https://zlib.net/zlib-1.2.11.tar.gz" -o zlib-1.2.11.tar.gz && \
+  echo "c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1  zlib-1.2.11.tar.gz" | shasum -a 256 -c - && \ 
+  tar -zxvf zlib-1.2.11.tar.gz && \
+  cd zlib-1.2.11 && \
+  ./configure --prefix=$PWD/install && \
+  make && make install && \
+  cd ..
+  
+RUN \  
+  curl -fsSL "https://github.com/libevent/libevent/releases/download/release-2.1.8-stable/libevent-2.1.8-stable.tar.gz" -o libevent-2.1.8-stable.tar.gz && \
+  curl -fsSL "https://github.com/libevent/libevent/releases/download/release-2.1.8-stable/libevent-2.1.8-stable.tar.gz.asc" -o libevent-2.1.8-stable.tar.gz.asc && \
+  gpg --keyserver pgp.mit.edu/ --recv 8EF8686D && \
+  gpg libevent-2.1.8-stable.tar.gz.asc && \
```

Comment:
Short key ids are forgeable -- should always use the full 160-bit fingerprint.  (...and maybe confirm that gpg _checks_ the full fingerprint; vague recollection it used to fail to check what it got.)

We should also commit the hash anyway even if we check the signature.  Update procedure: bump version, run build, _confirm signature passes and hash fails_, update hash.  If hash fails in any _other_ circumstances, something funny is afoot -- if not key compromise, then at least possible version substitution which the signature check doesn't catch (file name is not signed).

Reply:
The update procedure should also include checking the inside of the tarball to make sure the version number in the README or NEWS or CHANGELOG or whatever matches the advertised version, since the PGP signature does nothing to detect version rollbacks -- it doesn't cover the file name.

Reply:
Added `PGP` and `sha` checks for all.